### ISSUE TITLE
False positive in invariant_booleans

### ISSUE THREAD

**Issue by lexaknyazev** [dart-lang/linter#720]
Dart SDK 1.25.0-dev.3.0
```dart
void main() {
  var a = 0;
  var index = 0;
  var length = 5;
  switch(a) {
    case 0:
      while(index < length) {
        ++index;
      }
      break;
    case 1:
      while(index < length) { // <--------
        ++index;
      }
      break;
  }
}
```
gives (line 12):
```
info: Conditions should not unconditionally evaluate to "TRUE" or to "FALSE" verify: index < length. 
```

**Comment by alexeieleusis** [comment]
My guess is that the `while` statement in `case 0:` makes it trigger on `case 1`. Will look into this later, unless there is already a PR for it.

**Comment by kevmoo** [comment]
Just hit this with stageXL on 2.0.0-dev.3.0

**Comment by srawlins** [comment]
We've deprecated `invariant_booleans` and will not be fixing correctness bugs going forward.
